dojo/tools/npm_audit_7_plus/parser.py

```diff
@@ -0,0 +1,237 @@
+
+    def get_label_for_scan_types(self, scan_type):
+        """Return the scan label."""
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        """Return the scan description."""
+        return "NPM Audit Scan json output from v7 and above."
+
+    def get_findings(self, json_output, test):
+        """Return the findigs gathered from file upload."""
+        tree = self.parse_json(json_output)
+        return self.get_items(tree, test)
+
+    def parse_json(self, json_output):
+        """Parse the json format to get findings."""
+        if json_output is None:
+            return
+        try:
+            data = json_output.read()
+            try:
+                tree = json.loads(str(data, "utf-8"))
+            except Exception:
+                tree = json.loads(data)
+        except Exception:
+            raise ValueError("Invalid format, unable to parse json.")
+
+        # output from npm audit fix --dry-run --json
+        if tree.get("audit"):
+            if not tree.get("audit").get("auditReportVersion"):
+                raise ValueError(
+                    ("This parser only supports output from npm audit version"
+                        " 7 and above.")
+                )
+            subtree = tree.get("audit").get("vulnerabilities")
+        # output from npm audit --dry-run --json
+        # or
+        # output from npm audit --json
+        else:
+            if not tree.get("auditReportVersion"):
+                raise ValueError(
+                    ("This parser only supports output from npm audit version"
+                        " 7 and above.")
+                )
+            subtree = tree.get("vulnerabilities")
+
+        return subtree
+
+    def get_items(self, tree, test):
+        """Return the individual items found in report."""
+        items = {}
+
+        for key, node in tree.items():
+            item = get_item(node, tree, test)
+            unique_key = item.title + item.severity
+            items[unique_key] = item
+
+        return list(items.values())
+
+
+def get_item(item_node, tree, test):
+    """Return the individual Findigns from items found in report."""
+    vuln_ids = []
+    references = []
+    mitigation = ""
+    test = test
+    static_finding = True
+    title = ""
+    unique_id_from_tool = ""
+    cvssv3 = ""
+    cwe = ""
+
+    if item_node["severity"] == "low":
+        severity = "Low"
+    elif item_node["severity"] == "moderate":
+        severity = "Medium"
+    elif item_node["severity"] == "high":
+        severity = "High"
+    elif item_node["severity"] == "critical":
+        severity = "Critical"
+    else:
+        severity = "Info"
+
+    if item_node["via"] and isinstance(item_node["via"][0], str):
+        # this is a top level key (a vulnerability)
+        title = item_node["name"]
+        cwe = "CWE-1035"  # default
+        component_name = title
+        component_version = "n/a"
```

Comment:
No need to specify `n/a` here ([ref](https://documentation.defectdojo.com/contributing/how-to-write-a-parser/#not-all-attributes-are-mandatory))

Reply:
Thank you, addressed in this commit https://github.com/DefectDojo/django-DefectDojo/pull/9692/commits/31746fd05a45cc587dd51941ed680f5dff34a736